stdlib/wyvern/archast.wyv

```diff
@@ -0,0 +1,21 @@
+module archast
+
+type ComponentDecl	
+	def getType() : String
+	def getName() : String
+
+type PortDecl
+	def getProvides() : String
+	def getRequires() : String
+
+val NoneComponent : ComponentDecl = new
+	def getType() : String
+		"Nil"
+	def getName() : String
+		"Nil"
+
+val NonePort : PortDecl = new
+	def getProvides() : String
+		"Nil"
+	def getRequires() : String
+		"Nil"
```

Comment:
As I have mentioned above, the usage of NonePort in your code can be eliminated. Once you do that, I don't see why we need the NoneComponent and NonePort declarations here. Please remove them.